### PR TITLE
make read quota limiter work for coprocessor as well

### LINKED ISSUE #13256: QuotaLimiter: coprocessor read traffic should be controlled by foreground quota limiter's read bytes
### What version of TiKV are you using?
6.0 or later

### Steps to reproduce
set tikv config: quota.foreground_read_bytes: 5MB
run sysbench read-only workload

### What did you expect?
The coprocessor's read response size is close to 5MB/s per tikv
### What did happened?
The coprocessor's read response size is much larger than 5MB/s

### PR DESCRIPTION
**Pull request by tonyxuqqi**
Signed-off-by: qi.xu <tonxuqi@outlook.com>

### What is changed and how it works?

Issue Number: Close #13256

What's Changed:

```commit-message
The coprocessor's read bytes are not calculated by foreground quota limiter. 
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Manual test 

Side effects
- None

### Release note 

```release-note
None
```

### CODE REVIEW

File: components/tidb_query_executors/src/runner.rs [hunk 1]
```diff
@@ -471,6 +471,9 @@ impl<SS: 'static> BatchExecutorsRunner<SS> {
                     &mut ctx,
                 )?
             };
+            if chunk.has_rows_data() {
```
Comment (BornChanger) [inline components/tidb_query_executors/src/runner.rs:474]: Though I prefer to use following approach.  ForwardScanner maintains the statistics (via the Cursor interface), but it's lost before returning to internal_handle_request() in the test.  This patch is ok to me at present. 
image

File: tests/integrations/coprocessor/test_select.rs [hunk 2]
```diff
@@ -76,6 +77,7 @@ fn test_select() {
         let result_encoded = datum::encode_value(&mut EvalContext::default(), &row).unwrap();
         assert_eq!(result_encoded, &*expected_encoded);
     }
+    assert!(limiter.total_read_bytes_consumed(true) > 0); // the consume_sample is called due to read bytes quota
```
Comment (BornChanger) [inline tests/integrations/coprocessor/test_select.rs:80]: why not give a solid number?

File: components/tidb_query_executors/src/runner.rs (hunk 1 above)
Comment (tonyxuqqi) [reply components/tidb_query_executors/src/runner.rs:474]: note here the limiter is foreground limiter---that is to limits actual response size back to TiDB, not the IO limiter.

File: tests/integrations/coprocessor/test_select.rs (hunk 2 above)
Comment (tonyxuqqi) [reply tests/integrations/coprocessor/test_select.rs:80]: Here the logic is to verify if the read bytes based limiter is called. We don't know the actual response size here and so no way to get accurate wait time. 
For actual accurate wait time, it's covered by unit test.

**Review by BornChanger — APPROVED** [review]

**Review by ethercflow — APPROVED** [review]
LGTM

**Review by 5kbpers — APPROVED** [review]

**Pull request merged**